### PR TITLE
#2: Add general project description in english and german

### PR DESCRIPTION
**Pull request by florianrusch**
closes task `General project description in german and english` at story #2

### CODE REVIEW

File: docs/project_description.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+Usually, the various platforms and social media networks provide interfaces through which potential online profiles can be accessed. A further procedure is the scraping of information from the websites themselves.
+
+### Data aggregation
+
+The entered data is enriched with the obtained information in several runs and reused for a data collection. This process is repeated until no new data is collected.
+
+### Data presentation
+
+The information obtained is filtered and cleaned after data aggregation in order to make it accessible to the user.
```
Comment (ReySi) [inline docs/project_description.md]: **Enhancement/Wording**: The obtained information is filtered and cleaned up after the aggregation
**Question**: What do you mean exactly with cleaned and filtered and why is this necessary in order to visualize it?

File: docs/project_description.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+### Pattern for data analysis and aggregation
+
+- **Scoring** - weighting of information
+- **Image recognition** - extraction of information from images
+- **Video analysis** - extraction of information from Videos
+- **Face recognition** - matching and recognizing people in images and videos
+- **Text analysis** - extraction of information from texts
+- **Phonetic similarities** - recognition of similar spellings (e.g. Stephan, Steffan, Stefan)
```
Comment (ReySi) [inline docs/project_description.md]: **Question**: What's the difference between text analysis and phonetical similarities?

File: docs/project_description.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+- Online telephone directories
+- Newspapers
+- Search engines
+- ...
+
+## Data Protection/Legal Issues
+
+- Is it allowed to scrape public or personal data?
+- Is it allowed to scrape personal data from an API interface?
```
Comment (ReySi) [inline docs/project_description.md]: **Enhancement**: Consider adding a further description of why this could be a difficulty and the difference to the first point

File: docs/project_description.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+- Is it allowed to store the data on our servers?
+- Is it allowed to process the data on our servers?
+- Is it allowed to store the data on rented servers?
+- Is it allowed to process the data on rented servers?
```
Comment (ReySi) [inline docs/project_description.md]: **Enhancement**: Maybe external servers instead of rented servers?

File: docs/project_description_de.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    - [Datenquellen](#datenquellen)
+  - [Datenschutz/Rechtliche Fragestellungen](#datenschutzrechtliche-fragestellungen)
+
+## Motivation
+
+Heutzutage geben die meisten Menschen sämtliche Informationen über sich im Internet preis, ohne sich Gedanken darüber zu machen, wer alles auf diese Daten zugreifen kann. Inzwischen ist es möglich, anhand von ein paar wenigen Daten im Internet sehr viel über einen Menschen herauszufinden.
+
+## Ziel von _GoldenProfile_
+
+Mit dem Projekt _GoldenProfile_ soll ein Tool geschaffen werden, das auf Basis einiger weniger Stammdaten, wie zum Beispiel dem Namen und das Geburtsdatum einer Person, im Internet nach weiteren Daten sucht. Durchsucht werden hierbei neben den vorhandenen Social Media Netzwerken, auch Zeitungen und diverse weitere Internetplattformen. Ist der Crawling-Vorgang abgeschlossen, werden die Daten aufbereitet, aggregiert und dem Anwender zur Verfügung gestellt.
```
Comment (ReySi) [inline docs/project_description_de.md]: **Grammar**: Durchsucht werden hierbei **,** neben den vorhandenen Social Media Netzwerken.
**Wording**: Ist der Scraping-Vorgang abgeschlossen

File: docs/project_description_de.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+## Mögliche Use-Cases
+
+(Im Folgenden wird aus Gründen der besseren Lesbarkeit die männliche Form verwendet und ist in keiner Form wertend zu verstehen)
```
Comment (ReySi) [inline docs/project_description_de.md]: seriously? 😂

Comment (florianrusch) [reply docs/project_description_de.md]: Yes! ;-)

File: docs/project_description.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
+# _GoldenProfile_
```
[the remark this answers is not in the document]
Comment (florianrusch) [reply docs/project_description.md]: Good point. When I named the German version of the project description `project_description_de.md`, I thought about naming the English description similar.  But since we want to do the project completely in English, the "main language" of the documents should also be English.

I don't think we will have a German version for every document we create in the course of this project. For this reason, I think it would be better to have the main documents without a language suffix and any further version of the document with language suffix.

For example:
- architecture.md (English)
- project_description.md (English)
- project_description_de_md (German)
- structure.md (English)
- workflow.md (English)

Otherwise we would have this structure:
- architecture_en.md (English)
- project_description_en.md (English)
- project_description_de_md (German)
- structure_en.md (English)
- workflow_en.md (English)

There is always the case that documents are not updated correctly. Especially if you have the document in several languages.Therefore it's better to have a main document, which is always up-to-date (yes of course 😄 both documents should always be up-to-date. but how often is this not the case? Very often...)

File: docs/project_description.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+With the project _GoldenProfile_ a tool is to be created, which searches the Internet for further data on the basis of a few master data, such as the name and date of birth of a person. In addition to the existing social media networks, newspapers and various other Internet platforms are also searched. Once the crawling process is complete, the data is prepared, aggregated and made available to the user.
+
+## Possible Use Cases
+
+- A private individual wants to find out what can be found about him or her on the Internet.
+- An applicant wants to check what the HR department can find out about him or her.
+- An employee in the HR department wants to check an applicant and see what he or she has published on the Internet.
+- An employee in the HR department wants to know how sensitive an applicant is with his or her personal data.
+- An employee in the HR department wants to enrich the existing employee data with more detailed information.
```
[the remark this answers is not in the document]
Comment (florianrusch) [reply docs/project_description.md]: They are very similar, that is true. The **third point** is about the concrete data that has been published on the Internet. The **fourth point** is about the awareness of publishing data about oneself. But that could be elaborated further, that's right.

For example, you are in HR:
You work in the HR department of a company that has a lot of very secret data (e.g. BND, NAS, or whoever). Then you don't just want to know details about the person. You want to know how much the person is aware of what data he or she publishes about himself or herself on the Internet. If the person hides their birthday or address on any platform and has only posted a few pictures of themselves with some friends, you can be sure that they will handle the company data safely.

However, if the person publicly blasphemes about her old employer, you better shouldn't hire him or her.



**Fifth point**:
Yeah, you are right. But is there any difference if they are looking for Information about you in the application procedure or when you are already 5 years working for the company? But we could change the use case maybe and say that the companies don't extend the data, they just check from time to time what the employee publishes.

File: docs/project_description.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+- A key account manager prepares for a customer appointment with a new customer and requires a customer's professional and personal profile that is as comprehensive as possible.
+- A key account manager wants to enrich the existing information about a customer with further information.
```
[the remark this answers is not in the document]
Comment (florianrusch) [reply docs/project_description.md]: > Question: Isn't the fourth point basically the same use-case as the third one?
Also, I'm not sure if it's a good idea for an HR department to enrich existing employee data by information about the internet. Just imagine your company would search for data about you. I'd leave my company immediately, wouldn't you?

Isn't this the same question as in the above comment?

> The fifth one sounds pretty good to me but maybe we should only scrape their professional information here.

👍 Depending on the requirements. But I think this is a implementational decision and shouldn't appear on this point in this document. 😉

> **Enhancement/Wording**: The last one: a company wants to enrich their existing user data with available information about them on the Internet.

User = customer?

The Key Account Manager is the person who takes care of existing customers and is their contact person. He wants to have as much information about the customer as possible in order to take care of the customer and his needs as best as possible.

File: docs/project_description.md (hunk 1 above)
Comment (florianrusch) [reply docs/project_description.md]: Actually this are just buzzwords. I don't really know if we will have some data which have to be cleaned or filtered. This is quite possible and we should expect this.

File: docs/project_description.md (hunk 2 above)
Comment (florianrusch) [reply docs/project_description.md]: As I described it there: If we have some longer texts on a website or somewhere we have to filter some informations out of them.

The phonetical similarities could take a part of the text analysis. But I've understand also like it is a separate part of the search process: the user is inserting the name Steffan and we are also searching for the name Stephan. (The results of the search for Stephan should get in this context a lower scoring value as the results for Steffan)

File: docs/project_description.md (hunk 4 above)
Comment (florianrusch) [reply docs/project_description.md]: **Tradeoff**: [...] data on external rented servers?

File: docs/project_organization.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
+# Project management
```
Comment (florianrusch) [inline docs/project_organization.md]: Should I review that? Or do we just insert it into the repo and adjust it after the next meeting where we talk about it again?

File: README.md [hunk 11]
```diff
@@ -1 +1,9 @@
-GoldenProfile
```
Comment (florianrusch) [inline README.md]: I've updated this file in f4cffc778807b89808a08f59e0fee2a4f2bbae4b

File: README.md [hunk 12]
```diff
@@ -1 +1,12 @@
...
+# _GoldenProfile_
```
Comment (florianrusch) [inline README.md]: I've updated this file in f4cffc7

**Comment by florianrusch** [comment]
Sorry for this bunch of comments, reviews and other things I've generated here... I didn't know how I could comment/review @reysi's stuff on the best way (probably if we split `issue=story` into `issue=task` it will be very easy and that will never be a problem).

File: docs/project_organization.md (hunk 10 above)
Comment (ReySi) [reply docs/project_organization.md]: i'd suggest to readjust it after we talked about it
